### ISSUE TITLE
[Help Wanted] Re-implement `CTRL+SHIFT+S` as search in current channel

### ISSUE THREAD

**Issue by jasonblais** [mattermost/desktop#305]
#### Summary: Re-implement `CTRL+SHIFT+S` as search in current channel

This feature has already been implemented, but was reverted for v3.4 as the user experience was broken when used on direct message channels. 

`CTRL+SHIFT+S` auto-filled the searchbox with `in:[userid1_userid2]` for direct message channels, which was confusing and appeared as a bug to end users. 

For this feature to work properly, we'd need to submit a PR on the server side to change URLs of direct messages to usernames, that is `.../channels/username1_username2`. A Jira ticket is already available and marked as `accepting-pull-requests` https://mattermost.atlassian.net/browse/PLT-91 if anyone would like to take a look.

PS: Long-term plan with find/search functionality is to have
  - CTRL+S: search in current team (autofocus on search box)
  - CTRL+SHIFT+S: search in current channel
  - CTRL+F: browser-like find

**Comment by St-Ex** [comment]
Hi,
Don't forget that username may be changed by user. How do you think it will work for this scenario ?

**Comment by jasonblais** [comment]
Hey @St-Ex! 

The consideration was that the URL would in that case change to take the new username.

Given integrations are not supported in direct message channels and only the two users have access to the channel, I think that would be fine.

Would you feel that would work?

**Comment by St-Ex** [comment]
I think it could work but it might be conflicting with latest changes

> Channel URL no longer updates when the Channel Name changes

Now, when you change canal name, you may or not change the url

Ex : in our mattermost https://mattermost.bappli.com/studio/channels/it-rocks
![mattermost-changename](https://cloud.githubusercontent.com/assets/2870576/18916613/4573c100-8595-11e6-8968-26c9cce02151.png)

I think it might have given trouble to people using mattermost API.

**Comment by jasonblais** [comment]
Sorry @St-Ex for the very late reply here. This slipped by my radar.

You are correct that renaming a public/private channel no longer updates the URL. This was to prevent breaking those using APIs, integrations and public links to channels.

That being said, when you change your username, you're not just renaming a channel, you're updating your profile in Mattermost. It would be a little weird to have DMs where some URLs use the old username and some use the new one, I think they should all be consistent and use the new username. 

I've queued to discuss with our UX designer to confirm though, please let us know if you'd have further thoughts?

Thank you again for the feedback here, much appreciated!

**Comment by jasonblais** [comment]
@St-Ex This is what we arrived at

---- 
Cosmetically change URLs of Direct Message channels to be usernames instead of the concatenation of two user GUIDs

Specifically, the URL should be of the form ".../[TEAM NAME]/messages/[USERNAME]" where [USERNAME] is the username of the person you're talking to

If username is updated, the URL should cosmetically update to that new username.

*NOTE*: Internal name of the channel in the database should still be id1__id2
----

So in summary, the internal name remains `id1__id2`, but we would just change the URL cosmetically.

How would this sound @St-Ex?

**Comment by St-Ex** [comment]
Sounds good to me.

**Comment by jasonblais** [comment]
CTRL+S and CTRL+SHIFT+S is causing confusion among some users (example: #364), is inconsistent with the web app and doesn't follow typical conventions for searching or finding something.

Instead, we'll look into adding a browser-like find through CTRL+F shortcut (#399).

Hence, closing this ticket as "Won't Fix".
